src/engine/sparqlExpressions/StringExpressions.cpp

```diff
@@ -584,39 +584,37 @@ using LangMatches =
 
 // STRING WITH LANGUAGE TAG
 [[maybe_unused]] inline auto strLangTag =
-    [](std::optional<std::string> input,
+    [](std::optional<ad_utility::triple_component::Literal> literal,
        std::optional<std::string> langTag) -> IdOrLiteralOrIri {
-  if (!input.has_value() || !langTag.has_value()) {
+  if (!literal.has_value() || !langTag.has_value() ||
+      literal.value().hasLanguageTag() || literal.value().hasDatatype()) {
```

Comment:
does the `Literal` class have a `isPlain` function that checks exactly this (languageTag or datatype present, by simply checking startOfSuffix?
Use it here and implement it if not yet present to make the code simpler
(also below for the other function).